crates/aiken-project/src/blueprint/schema.rs

```diff
@@ -848,4 +1210,224 @@ pub mod test {
+        )
+    }
+
+    #[test]
+    fn deserialize_data_list_one() {
+        assert_eq!(
+            Data::List(Items::One(Declaration::Referenced(Reference::new("foo")))),
+            serde_json::from_value(json!({
+                "dataType": "list",
+                "items": { "$ref": "foo" }
+            }))
+            .unwrap()
+        )
+    }
+
+    #[test]
+    fn deserialize_data_list_many() {
+        assert_eq!(
+            Data::List(Items::Many(vec![
+                Declaration::Referenced(Reference::new("foo")),
+                Declaration::Referenced(Reference::new("bar"))
+            ])),
+            serde_json::from_value(json!({
+                "dataType": "list",
+                "items": [
+                  { "$ref": "foo" },
+                  { "$ref": "bar" }
+                ],
+            }))
+            .unwrap()
+        )
+    }
+
+    #[test]
+    fn deserialize_data_map() {
+        assert_eq!(
+            Data::Map(
+                Declaration::Referenced(Reference::new("foo")),
+                Declaration::Referenced(Reference::new("bar"))
+            ),
+            serde_json::from_value(json!({
+                "dataType": "map",
+                "keys": { "$ref": "foo" },
+                "values": { "$ref": "bar" }
+            }))
+            .unwrap()
+        )
+    }
+
+    #[test]
+    fn deserialize_any_of() {
+        assert_eq!(
+            Data::AnyOf(vec![Constructor {
+                index: 0,
+                fields: vec![
+                    Declaration::Referenced(Reference::new("foo")).into(),
+                    Declaration::Referenced(Reference::new("bar")).into()
+                ],
+            }
+            .into()]),
+            serde_json::from_value(json!({
+                "anyOf": [{
+                    "index": 0,
+                    "fields": [
+                        {
+                            "$ref": "foo",
+                        },
+                        {
+                            "$ref": "bar",
+                        }
+                    ]
+                }]
+            }))
+            .unwrap()
+        )
+    }
+
+    #[test]
+    fn deserialize_one_of() {
+        assert_eq!(
+            Data::AnyOf(vec![Constructor {
+                index: 0,
+                fields: vec![
+                    Declaration::Referenced(Reference::new("foo")).into(),
+                    Declaration::Referenced(Reference::new("bar")).into()
+                ],
+            }
+            .into()]),
+            serde_json::from_value(json!({
+                "oneOf": [{
+                    "index": 0,
+                    "fields": [
+                        {
+                            "$ref": "foo",
+                        },
+                        {
+                            "$ref": "bar",
+                        }
+                    ]
+                }]
+            }))
+            .unwrap()
+        )
+    }
+
```

Comment:
Some property-based test covering both `Schema` and `Data` serialization and deserialization. Makes sure that we can at least deserialize anything we serialize.

Reply:
epic